Various issue fixes + new modified indicator

itsschwer opened the pull request:
## changes
* Update `README.md` to list FloatFields and IntFields (added in v2.8.0) under "Currently supported options"
* Code-based fix for labels in ChoiceOptions appearing indented (addresses #49)
    * *my personal preference is to implement everything in code, but seeing as this project uses prefabs for UI elements, a better solution may be editing the actual prefab in Unity*

## additions
* Create `TestPlugin.cs` that is only built on Debug configurations for development/testing purposes.
    * *this can be deleted if it doesn't belong, but it won't exist on release builds anyway*
* Add a setting to `StepSliderConfig` to allow text entry of values that are not remapped to the nearest increment (addresses #30)
    * *I've named the setting `remapManualInputToStep`, which I'm <mark>**not sure is the best name**</mark> for this*
    * <mark>**Uncertain what the default value should be**</mark>. Currently uses `true` to match existing behaviour, making this change an opt-in
* Add a "modified" indicator to the side of options that have been modified *(similar to indicator in VS Code settings)*
    <img width="544" height="209" alt="modified-indicator" src="https://github.com/user-attachments/assets/f43c4061-3414-4f33-a3bf-01fc63adb7bb" />
    * Options that have a non-default value show a blue "modified" indicator
    * Options that have been modified and can be reverted using the "Revert" button *(now "Reset to Default" in Alloyed Collective update?)* show a yellow "modified" indicator. This takes precedence over the blue non-default indicator.
    * This is aimed at making it easier to identify which changes will be reverted or are non-default (and is slightly relevant to #27)
    * <mark>**Should there be a config option to disable showing this indicator?**</mark>
    * <mark>**Should the colours be changed/adjusted?**</mark> *(kinda chose them without too much thought)*

## closes
#30, #49
Given that the issue with 2.8.3 appears to have been resolved as [DiscordRichPresence](https://github.com/mikhailmikhalchuk/RoR2-Discord-RP/pull/12) accidentally bundling Risk of Options 2.8.2, a new release should also close the following: #44, #45, #46, #48

itsschwer committed:
readme: List new options added in 2.8.0

itsschwer committed:
Add remapManualInputToStep to StepSliderConfig

Addresses Rune580/RiskOfOptions#30

Name is a little bit long, not sure if a better name applies? (e.g ...ToIncrement might be more descriptive but also even longer)

itsschwer committed:
Create TestPlugin.cs

itsschwer committed:
Quick fix to choice option label position

Addresses Rune580/RiskOfOptions#49

itsschwer committed:
Add modified indicator to options

Add a blue bar to the left side of any option that does not have its default value (similar to VS Code settings)

Helps to identify which settings have been modified from their default value.

Potentially relevant to Rune580/RiskOfOptions#27

itsschwer committed:
Show modified indicator for revertable changes

itsschwer committed:
Separate out GetDefaultValue(); include warning

See Rune580/RiskOfOptions#28 ?

Rune580 commented:
Overall, I like these changes/improvements. However, I'd like to see the colors be configurable.

itsschwer commented:
Will do. I'll also add a config for enabling/disabling the modified indicator altogether?

itsschwer committed:
Add configs for modified indicator

Added a new public virtual member `ModSetting.UpdateModifiedIndicator` to allow refreshing all modified indicators when the configs are changed (so that enabled status and colour changes are applied live)

itsschwer commented:
Config options added.

To ensure that the modified indicator configs are applied as soon as the options change, I added a new `public virtual` method to `ModSetting` (`UpdateModifiedIndicator`), similar to how conditionally disabling mod settings is implemented (`CheckIfDisabled`).
